frontend/src/js/controllers/challengeCtrl.js

```diff
@@ -63,6 +64,30 @@
 
         utilities.showLoader();
 
+        vm.scrollToEntryAfterLeaderboardLoads = function () {
+            // get unique rank number from the url & if exists hightlight the entry
+            var elementId = $location.absUrl().split('?')[0].split('#')[1];
+            if (elementId) {
+                $anchorScroll(elementId);
+                $scope.currentEntryLink = elementId;
+                $anchorScroll.yOffset = 90;
+                $scope.isHighlight = elementId.split("leaderboardrank-")[1];
+            }
+        }
+      
+        // scroll to the specific entry of the leaderboard
+        vm.scrollToSpecificEntryLeaderboard = function (elementId) {
+            var newHash = elementId.toString();
+            if ($location.hash() !== newHash) {
+                $location.hash(elementId);
+                $scope.currentEntryLink = elementId;
+            } else {
+                $anchorScroll();
+            }
+            $scope.isHighlight = false;
+            $anchorScroll.yOffset = 90;
+        };
```

Comment:
Change this function to this:
```
vm.scrollToEntryAfterLeaderboardLoads = function () {
            // get unique rank number from the url & if exists hightlight the entry
            $timeout(function() {
                const elementId = $location.absUrl().split('?')[0].split('#')[1];
                console.log(elementId);
                if (elementId) {
                    $anchorScroll.yOffset = 90;
                    $anchorScroll(elementId);
                    $scope.currentEntryLink = elementId;
                    $scope.isHighlight = elementId.split("leaderboardrank-")[1];
                }
            }, 500);
        };
```

This is due this issue we are having: https://stackoverflow.com/a/30247770/4415561
and in this case it's okay to wait 500ms